### PR TITLE
Add Happy numbers problem in "others" category

### PR DESCRIPTION
**Pull request by louvelbr**
### **Describe your change:**

- [x] Add an algorithm? 
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References

Issue number : #2832

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.

### CODE REVIEW

**Comment by siriak** [comment]
Hi, the build is failing with the following error: ./src/main/java/com/thealgorithms/others/HappyNumbersSeq.java:4: error: class HappyNumSeqTest is public, should be declared in a file named HappyNumSeqTest.java
public class HappyNumSeqTest {
       ^
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

File: src/main/java/com/thealgorithms/searches/LinearSearchThread.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package search;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/LinearSearchThread.java:1]: Package names are not correct, please see correct ones in other files

File: src/main/java/com/thealgorithms/others/HappyNumbersSeq.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		System.out.print("Enter number: ");
+		int n = in.nextInt();
+		while(n != 1 && !isSad(n)){
+			System.out.print(n+" ");
+			n = sumSquares(n);
+		}
+		if(n == 1) System.out.println("1 Happy number");
+		else
+			System.out.println("Sad number");
+	}
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/HappyNumbersSeq.java:16]: Having some formatting would be great

**Review by siriak — CHANGES_REQUESTED** [review]

**Review by yanglbme — APPROVED** [review]

**Comment by yanglbme** [comment]
Hi @siriak , please review again.

File: src/main/java/com/thealgorithms/searches/LinearSearchThread.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        this.x = x;
+    }
+
+    @Override
+    public void run() {
+        int k = a;
+        found = false;
+        while (k < b && !found) {
+            if (f[k] == x) {
+                found = true;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/LinearSearchThread.java:57]: Please short-circuit execution here, no need to continue iterations

File: src/main/java/com/thealgorithms/searches/LinearSearchThread.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    private int[] f;
+    private int a, b;
+    private int x;
+    private boolean found;
+
+    Searcher(int[] f, int a, int b, int x) {
+        this.f = f;
+        this.a = a;
+        this.b = b;
+        this.x = x;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/LinearSearchThread.java:48]: Please improve naming. array, leftIndex, rightIndex,...

**Review by siriak — COMMENTED** [review]
Only a couple of small comments, everything else looks good :)

**Review by siriak — APPROVED** [review]

**Pull request merged**